### PR TITLE
Update FontoXPath to its latest version

### PR DESCRIPTION
**Pull request by DrRataplan**
This introduces the ALL_RESULTS type, which in turn makes XPath more predictable. This commits alos
cleans up most of the `Array.isArray(results) ? [results] : results` mess that the `ANY_TYPE` required.

### CODE REVIEW

**Comment by DrRataplan** [comment]
This will fix #127

**Comment by JoernT** [comment]
shouldn't there be the change of package.json also? 

I still have to admit that i don't understand all of the changes - i'll leave comments...

File: src/ForeElementMixin.js [hunk 1]
```diff
@@ -135,7 +137,7 @@ export const foreElementMixin = superclass =>
         if (nodeType) {
           this.nodeset = evaluateXPathToFirstNode(this.ref, inscopeContext, this);
         } else {
-          this.nodeset = evaluateXPath(this.ref, inscopeContext, this);
+          [this.nodeset] = evaluateXPath(this.ref, inscopeContext, this);
```
Comment (JoernT) [inline src/ForeElementMixin.js:140]: so the return of evaluateXPath will always be an array now - do i get this right?

**Review by JoernT — APPROVED** [review]

**Comment by JoernT** [comment]
re-reading helped

**Pull request merged**